packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts

```diff
@@ -148,5 +160,72 @@ test.describe('PieCookieBanner - Component tests', () => {
 
         expect(modal).not.toBeVisible();
         expect(cookieBanner).not.toBeVisible();
+
+        const [expectedCookieBannerPrefsSavedEvent] = preferences.filter(({ id }) => id !== 'all')
+        .map(({ id, isChecked }) => ({ [id]: !!isChecked }));
+        expect(cookieBannerPrefsSavedEvent).toMatchObject(expectedCookieBannerPrefsSavedEvent);
+    });
+
+    test('should always set the `necessary` preference to on and disabled states', async ({ mount, page }) => {
+        // Arrange
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+        });
+
+        // Act
+        await page.click(managePreferencesSelector);
+
+        // Assert
+        const isChecked = await page.locator(getPreferenceItemSelector('necessary')).isChecked();
+        const isDisabled = await page.locator(getPreferenceItemSelector('necessary')).isDisabled();
+
+        expect(isChecked).toBe(true);
+        expect(isDisabled).toBe(true);
+    });
+
+    test('should turn on all preferences if the `all` preference is set to true', async ({ mount, page }) => {
+        // Arrange
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+        });
+
+        // Act
+        await page.click(managePreferencesSelector);
+
+        // Assert
+        await page.click(getPreferenceItemSelector('all'));
+
+        const elements = preferences.map(async ({ id }) => ({
+            isChecked: await page.locator(getPreferenceItemSelector(id)).isChecked(),
+        }));
+
+        const results = await Promise.all(elements);
+
+        results.forEach((result) => {
+            expect(result.isChecked).toBe(true);
+        });
+    });
+
+    test('should turn off all preferences if the `all` preference is set to false except for the necessary preference', async ({ mount, page }) => {
```

Comment:
Do you need to make sure that the other toggles are actually set to false here? Right now it only checks that the necessary one has a truthy value?

Reply:
Hi ~
it actually does check if the other elements are set to false. 

we are looping over items and making sure only if `id === 'necessary'` is true, and others are false. 

I tried to be explicit such as `id === 'necessary' ? true : false` but eslint refactors it to `id === 'necessary'` only as it indirectly implies that. 

```
  results.forEach(({ id, isChecked }) => {
            expect(isChecked).toBe(id === 'necessary');
        });
```